src/cloud/functions/leaderboards/around-me/leaderboard/run.csx

```diff
@@ -0,0 +1,240 @@
+    if (runOnce)
+    {
+        log.Info("Running initialization");
+
+        if (string.IsNullOrWhiteSpace(baseArchitectureVersion) ||
+            !baseArchitectureVersion.StartsWith(requiredBaseArchitectureVersion)) log.Error($"The base architecture version doesn't match the expected version {requiredBaseArchitectureVersion}");
+
+        // Check required application settings
+        if (string.IsNullOrWhiteSpace(db)) log.Error("COSMOSDB_DBNAME settings wasn't provided");
+        if (string.IsNullOrWhiteSpace(endpoint)) log.Error("COSMOSDB_ENDPOINT settings wasn't provided");
+        if (string.IsNullOrWhiteSpace(key)) log.Error("COSMOSDB_PRIMARY_MASTER_KEY settings wasn't provided");
+
+        // Create Cosmos DB Client from settings
+        client = new DocumentClient(new Uri(endpoint), key);
+
+        // Create Database and Collection in Cosmos DB Account if they don't exist
+        await client.CreateDatabaseIfNotExistsAsync(new Database { Id = db });
+        await client.CreateDocumentCollectionIfNotExistsAsync(
+            UriFactory.CreateDatabaseUri(db),
+            new DocumentCollection { Id = collection });
+
+        runOnce = false;
+
+        log.Info("Initialization done!");
+    }
+
+    if (string.IsNullOrWhiteSpace(leaderboard) || string.IsNullOrWhiteSpace(playerId))
+        return req.CreateResponse(HttpStatusCode.BadRequest, "Both playerId and leaderboard should be provided in querystring, format is leaderboard/playerId");
+
+    try
+    {
+
+        var scores = client.CreateDocumentQuery<ScoreItem>(
+            UriFactory.CreateDocumentCollectionUri(db, collection), new FeedOptions { EnableCrossPartitionQuery = true });
+
+        // Get score for the current player by playerId        
+        var query =
+            (from s in scores
+             where s.PlayerId == playerId && s.Leaderboard == leaderboard
+             select s);
+
+        var player = query.ToList();
+        if (player.Count < 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"Player with Id = {playerId} in Leaderboard = {leaderboard} does not exist in the database");
+
+        if (player.Count > 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"There are more than one player with Id = {playerId} in Leaderboard = {leaderboard}. This is DB confuguration error - this combination should be unique");
+
+        var currentPlayer = player.First();
+
+        // Calculate global rank for the current player by playerId 
+        int globalRank;
+
+        query = from c in scores
+                where c.Score > currentPlayer.Score && c.Leaderboard == leaderboard
+                select c;
+
+        List<LeaderboardItem> sortedResult = new List<LeaderboardItem>(); //Sorted list from highest score (start of list) to lowest for output  
+
+        globalRank = query.Count<ScoreItem>() + 1;
+
+        if (globalRank > 1) //if current player it not number 1 in global rank
```

Comment:
WOW! :-) Very creative solution. I think we can simplify this A LOT! The problem is in fact the same as what has been solved in the "top-n scenario" but you just have to execute the loop backwards and forwards for people with a rank ahead of you and for the people with the same rank or below you.

At this point in the code, we know the rank of the current player and we can ask DocumentDB to retrieve a small set of people that have scores that are > than my score with the Take(n) option. And the same for people that have the same score or < than my score with the Take(n) option. Make sure those results are sorted accordingly, then just make a clean loop through them and implement the same algorithm as in the "top-n scenario" and you'll be done :-) Yes it's more complicated, but the problem scope is very similar.

Reply:
made some significant simplification but still it is more complex than forward loop